app/models/translation.rb

```diff
@@ -8,40 +8,32 @@ class Translation < ApplicationRecord
   include NotifierConfig
 
   def self.translate(text)
-    Rails.cache.fetch("translations/#{text}", expires_in: 15.minutes) do
-      translations = all.pluck(:key, :text).to_h
-      translations.each do |k, v|
-        Rails.cache.write("translations/#{k}", (v.presence || k), expires_in: 15.minutes)
+    # don't set expiry, the cache is updated via BuildTranslationCacheJob
+    translated = Rails.cache.fetch(cache_key(text)) do
+      translation = where(key: text).order(:id).first_or_create do |record|
+        record.key = text
+        Rails.logger.info("Unknown Translation key \"#{text}\", added to DB")
       end
-      translated = translations[text]
-      if ! translations.key?(text)
-        msg = "Unknown Translation key \"#{text}\", added to DB"
-        # Fail painfully if this is development so we see the error of our ways
-        # Actually, don't fail because sometimes we don't know where the string is coming from (pagy)
-        # raise msg if Rails.env.development?
-
-        # Notify if this isn't development so we can track it down later
-        # send_single_notification(msg, 'Translations', exception: StandardError.new(msg))
-
-        # Add it to the database if we don't have it
-        Translation.where(key: text).first_or_create
-        Rails.logger.info(msg)
-      end
-
-      translated.presence || text
+      translation.text
     end
+    translated || text
```

Comment:
Pretty sure we want 
```suggestion
    translated.presence || text
```

Otherwise if we delete the value from the UI (to restore the original) you'll get an empty string instead.
```
irb(main):001:0> "" || 'first'
=> ""
```

Reply:
Good catch, thanks!